docker/Dockerfile.dev

```diff
@@ -75,86 +94,64 @@ RUN chmod +x /tini
 RUN apt-get purge -y --auto-remove gnupg apt-transport-https
 
 RUN locale-gen en_US.UTF-8
-ENV LC_ALL en_US.UTF-8
-ENV LANG en_US.UTF-8
-ENV LANGUAGE en_US.UTF-8
+ENV LC_ALL=en_US.UTF-8 \
+    LANG=en_US.UTF-8 \
+    LANGUAGE=en_US.UTF-8
 
 RUN adduser app sudo; \
     echo '%sudo ALL=(ALL) NOPASSWD:ALL' >> /etc/sudoers;
 
-COPY . /app/wpt-sync
-RUN chown -R app:app /app/wpt-sync
-
 COPY ./config/rabbitmq.conf /etc/rabbitmq/rabbitmq.conf
 RUN echo LOG_BASE=/app/workspace/logs/rabbitmq >> /etc/rabbitmq/rabbitmq-env.conf
 
+COPY ./docker/start_wptsync.sh /app/start_wptsync.sh
+COPY ./docker/get_ini.py /app/get_ini.py
+RUN chmod +x /app/get_ini.py /app/start_wptsync.sh
+
 USER app
 
 # Set up files and config
-RUN mkdir /app/workspace \
+RUN mkdir -p /app/workspace \
     /app/workspace/logs \
     /app/workspace/logs/rabbitmq \
-    /app/testdata \
+    /app/wpt-sync \
     /app/repos \
-    /app/git-cinnabar \
-    /app/fzf \
-    /app/.ssh
+    /app/third_party/git-cinnabar \
+    /app/third_party/fzf \
+    /app/.ssh \
+    /app/requirements \
+    /app/.ipython/profile_default/startup/
 
 RUN sudo chown -R rabbitmq:rabbitmq /app/workspace/logs/rabbitmq
 
-ENV RABBITMQ_CONFIG_FILE=/etc/rabbitmq/rabbitmq
-
-COPY ./config/ssh_config /app/.ssh/config
+COPY ./config/ssh_config ./config/known_hosts /app/.ssh/
 RUN sudo chown -R app:app /app/.ssh/
 
 # Install git-cinnabar
-WORKDIR /app/git-cinnabar
-ENV PATH=/app/git-cinnabar:$PATH
-
-RUN git clone https://github.com/glandium/git-cinnabar.git . \
-    && git checkout origin/release; \
-    pip install --user requests \
+RUN cd /app/third_party/git-cinnabar \
```

Comment:
Distinct WORKDIR commands are preferred over `RUN cd ... && ...`

I think it's mostly a readability/style recommendation, but also each RUN command adds a new image layer so you want to use RUN for instructions that actually change the image.

Reply:
In this case we need to change the image anyway (since we actually need to install stuff) and we only care about the directory for the duration of the RUN command, so it seems less stateful to set it inside that command and not change the global state of the image. I think if we were just RUNing a single line of script here I'd be more inclined to go with seperate commands, but as it is, I don't think it actually is clearer, in spite of the best practice.